components/server/src/user/user-controller.ts

```diff
@@ -151,6 +152,21 @@ export class UserController {
                     // We respond with NOT_AUTHENTICATED to prevent gleaning whether the user, or token are invalid.
                     throw new ResponseError(ErrorCodes.NOT_AUTHENTICATED, "Admin user not found");
                 }
+
+                // Ensure admin user is owner of any Org.
+                const { rows: orgs } = await this.teamDb.findTeams(
+                    0 /* offset */,
+                    1000 /* limit */,
+                    "creationTime" /* order by */,
+                    "ASC",
+                    "" /* empty search term returns any */,
+                );
+                for (const org of orgs) {
```

Comment:
Would it not be better to create an admin resource guard that allows any operations? OR alternatively filter out the admin from the members list? We already do that in quite a few places.

Reply:
> admin resource guard that allows any operations

Those resource guards are to be replaced by the new permissions database. So each of those proposals are temporary. 

> alternatively filter out the admin from the members list

That's an option indeed. Having the special cases for the hidden admin is a nasty workaround for the time where the  proper authorization and service accounts aren't in place.

The pros of explicit removal/adding of the admin user to the Org:
  * it makes it even visible, that the "rescue mode" is actually used, otherwise w/o any audit logs available, it would be unnoticeable.
  * this intermediate state, under the assumption that it will be redone with service account permissions eventually, is centralized, and can be removed easier, while adding exceptional handling in different component produces more tech-debt.

Reply:
Ok, while not clean I agree the real problem is the special admin user object. Can we set the limit to 1? So it is more explicit that this if for the single-org situation? Seems weird to update the first 1000 orgs.

> Having the special cases for the hidden admin is a nasty workaround for the time where the proper authorization and service accounts aren't in place.

Let's not generally hold our breath and add technical-dept because we might have a shiny solution in a few weeks. We could solve this using the existing authorization but that wouldn't be much better as you pointed out.